File: python/credit_check.py
singles = []

def credit_check(str):

    for num in str[-1::-2]:
        singles.append(int(num))

    for num in str[-2::-2]:
        num = int(num) * 2

        if num < 9:
            singles.append(num)
        else:
            num = ((num //10) + (num %10))
            singles.append(num)

    if (sum(singles)) % 10 == 0:
        return "The number is valid!"
    
    else:
        return "The number is invalid!"

# print(credit_check("4024007136512380"))
# print(credit_check("6011797668867828"))

# print(credit_check("4024007106512380"))    
# print(credit_check("6011797668868728"))
# print(credit_check("6011797668868728") == "The number is invalid!")

File: js/creditCheck.js
let singles = [];

function creditCheck(str) {
    for (let i = str.length - 1; i >= 0; i -= 2) {
        singles.push(parseInt(str[i]));
    }

    for (let i = str.length - 2; i >= 0; i -= 2) {
        let num = parseInt(str[i]) * 2;

        if (num < 9) {
            singles.push(num); 
        } else {
            num = Math.floor(num / 10) + (num % 10);
            singles.push(num);
        }
    }
    let sum = 0;
    singles.forEach(num => {
        sum += num;
    })

    if (sum % 10 == 0) {
        return "The number is valid!"
    } else {
        return "The number is invalid!"
    }
}

module.exports = creditCheck;